lib/translator.js

```diff
@@ -1,80 +1,167 @@
+    this.pattern = options.pattern || DEFAULT_REGEXP;
+    this.patternSplitter = options.patternSplitter || '|';
+    this.userTransform = options.transform || {};
+
+    if (typeof options === 'string'){
+      var lang = options.match(/([\.^\/]*)\.\w{0,4}$/);
+      this.lang = lang && lang[0] || 'undefined';
+      this.localePath = options;
+    }
+    else {
+
+      this.lang = options.lang;
+      this.localePath = options.localePath;
+      this.localeDirectory = options.localeDirectory;
+      this.localeExt = options.localeExt;
+
+      if (!this.localePath){
+        this.localePath = this.localeDirectory + this.lang;
+      }
+
+      if (this.localeExt) {
+        this.localePath += this.localeExt;
+      }
+    }
+
+    this.dictionary = getDictionary(this.localePath);
+
+    return this;
   };
 
   Translator.prototype.translate = function(content) {
     var resultPromise = q.defer();
     var self = this;
 
-    var localeName = Object.keys(this.locale)[0];
-    var localeValues = this.locale[localeName];
-
-    var expressions = content.match(EXPRESSION_REG);
-
-    if(expressions) {
-      resultPromise.resolve(expressions.reduce(function(content, expression) {
-        var filters = expression.match(FILTER_REG)[0].split('|').slice(1).map(function(key){
-          return key.trim();
-        });
-
-        var expressionValue = getLocaleValue(
-          expression.match(EXPRESSION_VALUE_REG)[1],
-          localeValues
-        ) || resultPromise.reject("Cannot find that key in locales " + expression);
-
-        var resultValue = filters.reduce(function(value, filter){
-          if(self[filter] && typeof self[filter] === 'function') {
-            return self[filter](value);
-          } else {
-            return resultPromise.reject(filter + " filter is not supported");
-          }
-        }, expressionValue);
-
-        return content.replace(
-          expression,
-          resultValue
-        );
-      }, content));
-    } else {
-      resultPromise.resolve(content);
-    }
+    resultPromise.resolve(content.replace(self.pattern, function (s, data) {
+
+      var transforms = data.split(self.patternSplitter).map(function(filter){
+        return filter.trim();
+      });
+
+      var value = transforms.splice(0, 1)[0];
+
+      var res = transforms.reduce(function(res, transformName){
+        if (res instanceof Error) {
+          return res;
+        }
+        if (typeof self.userTransform[transformName] === 'function') {
+          return self.userTransform[transformName](res, self.dictionary);
+        }
+        else if (typeof baseTransforms[transformName] === 'function') {
+          return baseTransforms[transformName](res, self.dictionary);
+        } else {
+          return new Error('No such transform: '+ transformName);
+        }
+      }, value);
+
+      if (res instanceof Error) {
+        return resultPromise.reject(res);
+      }
+     // console.log('c', content);
```

Comment:
Should i remove comments? Or something else? Can\'t understand GitHub line lines. Does it show commented line as the last in code window?